### ISSUE TITLE
wxOwnerDrawnComboBox: wrong selection when two items have the same name but are written with lower and upper case

### ISSUE THREAD

**Issue by vwichmann** [wxWidgets/wxWidgets#24289]
#### Bug description:

If there are two items with the same name in the choice list of a wxOwnerDrawnComboBox, but one is written upper case and the other one lower case, always the first in the list gets selected (second one can't be selected). It works fine when both items are written either upper or lower case.

This relates to an already fixed [issue](https://github.com/wxWidgets/wxWidgets/issues/19289). The fix allowed to correctly select two items with the same name when both are written either lower or upper case. Sorry that we missed to test the upper/lower combination by that time.

#### Expected vs observed behaviour:

It should be possible to select each item in a list, regardless how it is written and if it is a duplicate name or not.

#### Patch or snippet allowing to reproduce the problem:

Attached is a patch to the combo sample which allows to reproduce the problem. It simply replaces three item names with "Test", "Test", and "test". 

[repro-for-combo-selection-bug.txt](https://github.com/wxWidgets/wxWidgets/files/14178685/repro-for-combo-selection-bug.txt)

#### To Reproduce:

1. apply the patch to the combo sample and run it
2. try to select the "Test/Test/test" items from the "OwnerDrawnComboBox with owner-drawn items" list:
3. Select the first upper case "Test" item in the list, log is "EVT_COMBOBOX(id=-31991,selection=6)" and the item is correctly selected
4. Select the second upper case "Test" item in the list, log is "EVT_COMBOBOX(id=-31991,selection=7)" and the item is correctly selected
5. Select the lower case "test" item in the list, log is "EVT_COMBOBOX(id=-31991,selection=8)", but although "selection=8" is reported, the first upper case "Test" item gets selected

### Platform and version information

- wxWidgets version: git master

**Comment by vwichmann** [comment]
Just noticed that the selection works fine in the sample with "OwnerDrawnComboBox with simple dropbottom graphics" which uses wxOwnerDrawnComboBox directly.

It fails with "OwnerDrawnComboBox with owner-drawn items" and "OwnerDrawnComboBox with owner-drawn items and buttom on the left", both being created with the sample's wxPenStyleComboBox derived from wxOwnerDrawnComboBox.

**Comment by vadz** [comment]
Thanks for reporting this!

We must be (wrongly) using case-insensitive comparison somewhere, if you can find where exactly does it happen, it should be quite simple to fix. Please try debugging it if you can in order to do it. TIA!

**Comment by vwichmann** [comment]
The attached patch uses a case sensitive string comparison in wxVListBoxComboPopup::FindItem(). With the patch applied, case sensitive selections work fine in the combo sample. I don't know if this change could have any unwanted side-effects in other places.

[fix_combo_selection_case_sensitive.txt](https://github.com/wxWidgets/wxWidgets/files/14191236/fix_combo_selection_case_sensitive.txt)

**Comment by vwichmann** [comment]
I submitted a pull request for my proposed fix, but making FindItem() case sensitive now makes one of the tests fail:

~~~
D:/a/wxWidgets/wxWidgets/tests/controls/ownerdrawncomboboxtest.cpp(185): FAILED:
  REQUIRE( "item 2" == m_combo->GetValue() )
with expansion:
  "item 2" == "item 1"
~~~

In the test there is a comment that FindItem() is case insensitive, and this is checked by:

~~~
// Since this uses FindString it is case insensitive
m_combo->SetValue("ITEM 2");

CPPUNIT_ASSERT_EQUAL("item 2", m_combo->GetValue());
~~~

So it seems like this is by design. If this is the case, this issue should probably be closed.

**Comment by vwichmann** [comment]
I think it would be nice to have a case sensitive selection option as this is probably what most users would expect, but I'm afraid that I will not be able to implement your suggestion on my own.

**Comment by vwichmann** [comment]
Yes, most likely a backward compatible solution is required. As this seems to be quite complicated, please feel free to close this issue. Would be a nice feature, but it is not essential to me. Thanks for your help!
